packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2431,6 +2431,7 @@ class InputDecoration {
     this.errorStyle,
     this.errorMaxLines,
     this.hasFloatingPlaceholder = true,
+    this.alwaysFloatLabel = false,
```

Comment:
I wonder if this might be better as an enum like `floatLabel`.  Angular seems to also have the option to "never" float the label, which I think can't be done in Flutter right now either.  An enum would keep this option open in the future.  See https://material.angular.io/components/form-field/examples.

What do you think?

Reply:
Yeah great idea, sounds good.